src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/search/crosswikisearch/CrossWikiSearchPageObject.java

```diff
@@ -182,13 +183,26 @@ public void verifyNoResultsCaption() {
                           true);
   }
 
-  public void verifyThumbnails(int number) {
+  public void verifyThumbnailsAmount(int number) {
     Assertion.assertNumber(thumbnails.size(), number, "checking number of thumbnails");
+    PageObjectLogging.log("verifyThumbnailsAmount", "thumbnails verified",
+                          true);
+  }
+
+  public boolean areThumbnailsContainImages() {
+    String jpgOrPngImage = ".*\\.(png|jpg).*";
     for (WebElement elem : thumbnails) {
-      Assertion.assertStringContains(elem.getAttribute("src"), ".png");
+      boolean isImage = Pattern.matches(jpgOrPngImage, elem.getAttribute("src"));
+      try {
+        if (!isImage) {
+          throw new AssertionError();
+        }
+      } catch (AssertionError ass) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)